### DISCUSSION TITLE
What do the different pipe colors mean in SCIRun?

### DISCUSSION THREAD

**Discussion by tpat95** [SCIInstitute/SCIRun#2190] [Q&A] [answered]
Is there a tutorial video that we could reference to better understand these things?

**Comment by jab0707** [comment, marked as answer]
Yellow: Field - A geometry that includes nodes (minimum) as well as connectivity (faces/cells/elements), and data (either on the nodes (linear basis) or on the elements (constant basis)
Blue: Matrix -  data, either 1D or 2D matrices
Green: String
Pink: View - Things being passed into a view scene module
Purple: Color maps - color map information
Orange: Bundles - a group of other data types including fields and matrices
Gray: Programmable - used with python interfacing modules to control when code form module slike loop start/loop end is executed
